From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Wed, 15 Apr 2026 15:02:22 +0200
Subject: global-shortcuts-provider: Relax app-id validation

Snap app identifiers do not follow the GLib convention of being DBus names,
that is reverse-DNS style with at least two elements.
Relax our validation to allow snaps to use the global shortcuts portal; we only
really need to check that the app-id is not NULL.

Forwarded: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3382
---
 global-shortcuts-provider/cc-global-shortcuts-provider.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/global-shortcuts-provider/cc-global-shortcuts-provider.c b/global-shortcuts-provider/cc-global-shortcuts-provider.c
index 2513773..4a3be42 100644
--- a/global-shortcuts-provider/cc-global-shortcuts-provider.c
+++ b/global-shortcuts-provider/cc-global-shortcuts-provider.c
@@ -102,7 +102,7 @@ handle_bind_shortcuts (CcGlobalShortcutsProvider *self,
 {
   g_autoptr(CcGlobalShortcutDialog) shortcut_dialog = NULL;
 
-  if (!g_application_id_is_valid (app_id))
+  if (!app_id || !app_id[0])
     {
       g_warning ("Discarded shortcut bind request from application with an invalid app_id >%s<.", app_id);
       return G_DBUS_METHOD_INVOCATION_UNHANDLED;
